tests/unit/lms/services/lti_grading/factory_test.py

```diff
@@ -0,0 +1,56 @@
+from unittest.mock import Mock, sentinel
+
+import pytest
+
+from lms.services.lti_grading.factory import factory
+
+
+class TestFactory:
+    def test_v11(
+        self,
+        pyramid_request,
+        application_instance_service,
+        LTI11GradingService,
+        http_service,
+        oauth1_service,
+    ):
+        application_instance_service.get_current.return_value = Mock(lti_version="1.1")
+
+        svc = factory(sentinel.context, pyramid_request)
+
+        LTI11GradingService.assert_called_once_with(
+            sentinel.grading_url, http_service, oauth1_service
+        )
+        assert svc == LTI11GradingService.return_value
+
+    def test_v13(
+        self,
+        pyramid_request,
+        LTI13GradingService,
+        application_instance_service,
+        ltia_http_service,
+    ):
+        application_instance_service.get_current.return_value = Mock(
+            lti_version="1.3.0"
+        )
+        svc = factory(sentinel.context, pyramid_request)
+
+        LTI13GradingService.assert_called_once_with(
+            sentinel.grading_url, ltia_http_service
+        )
+        assert svc == LTI13GradingService.return_value
```

Comment:
Yeah. These are kind of long winded, but only because we have long names for our services. I don't think you can do this any better.